google/cloud/internal/parse_rfc3339.cc

```diff
@@ -14,218 +14,25 @@
-  return std::chrono::system_clock::from_time_t(std::mktime(&tm));
-}
-
-std::chrono::system_clock::duration ParseFractionalSeconds(
-    char const*& buffer, std::string const& timestamp) {
-  if (buffer[0] != '.') {
-    return std::chrono::system_clock::duration(0);
-  }
-  ++buffer;
-
-  long fractional_seconds;  // NOLINT(google-runtime-int)
-  int pos;
-  auto count = std::sscanf(buffer, "%9ld%n", &fractional_seconds, &pos);
-  if (count != 1) {
-    ReportError(timestamp, "Invalid fractional seconds component.");
-  }
-  auto constexpr kMaxNanosecondDigits = 9;
-  auto constexpr kNanosecondsBase = 10;
-  // Normalize the fractional seconds to nanoseconds.
-  for (int digits = pos; digits < kMaxNanosecondDigits; ++digits) {
-    fractional_seconds *= kNanosecondsBase;
-  }
-  // Skip any other digits. This loses precision for sub-nanosecond timestamps,
-  // but we do not consider this a problem for Internet timestamps.
-  buffer += pos;
-  while (std::isdigit(buffer[0]) != 0) {
-    ++buffer;
-  }
-  return std::chrono::duration_cast<std::chrono::system_clock::duration>(
-      std::chrono::nanoseconds(fractional_seconds));
-}
-
-std::chrono::seconds ParseOffset(char const*& buffer,
-                                 std::string const& timestamp) {
-  if (buffer[0] == '+' || buffer[0] == '-') {
-    bool positive = (buffer[0] == '+');
-    ++buffer;
-    // Parse the HH:MM offset.
-    int hours, minutes, pos;  // NOLINT(readability-isolate-declaration)
-    auto count = std::sscanf(buffer, "%2d:%2d%n", &hours, &minutes, &pos);
-    auto constexpr kExpectedOffsetWidth = 5;
-    auto constexpr kExpectedOffsetFields = 2;
-    if (count != kExpectedOffsetFields || pos != kExpectedOffsetWidth) {
-      ReportError(timestamp, "Invalid timezone offset, expected [+-]HH:MM.");
-    }
-    if (hours < 0 || hours >= kHoursInDay) {
-      ReportError(timestamp, "Out of range offset hour.");
-    }
-    if (minutes < 0 || minutes >= kMinutesInHour) {
-      ReportError(timestamp, "Out of range offset minute.");
-    }
-    buffer += pos;
-    using std::chrono::duration_cast;
-    if (positive) {
-      return duration_cast<std::chrono::seconds>(std::chrono::hours(hours) +
-                                                 std::chrono::minutes(minutes));
-    }
-    return duration_cast<std::chrono::seconds>(-std::chrono::hours(hours) -
-                                               std::chrono::minutes(minutes));
-  }
-  if (buffer[0] != 'Z' && buffer[0] != 'z') {
-    ReportError(timestamp, "Invalid timezone offset, expected 'Z' or 'z'.");
-  }
-  ++buffer;
-  return std::chrono::seconds(0);
-}
-
-#include "google/cloud/internal/diagnostics_pop.inc"
-}  // anonymous namespace
+#include "absl/strings/ascii.h"
+#include "absl/time/time.h"
 
 namespace google {
 namespace cloud {
 inline namespace GOOGLE_CLOUD_CPP_NS {
 namespace internal {
 std::chrono::system_clock::time_point ParseRfc3339(
```

Comment:
I also updated the header to remove the specific reference to "Google Cloud Storage" with ...
```
 * Google Cloud libraries often use RFC-3339 for timestamps. This function is
 * used to parse them and convert to `std::chrono::system_clock::time_point`,
 * the C++ class used to represent timestamps. Depending on the underlying
 * C++ implementation the timestamp may lose precision. C++ does not specify
 * the precision of the system clock, though most implementations have sub-
 * second precision, and nanoseconds is common. The RFC-3339 spec allows for
 * arbitrary precision in fractional seconds, though it would be surprising
 * to see femtosecond timestamp for Internet events.
```

Reply:
Done.